src/assertion/mod.rs

```diff
@@ -83,32 +126,127 @@ pub enum AssertionResult {
     Failed,
     /// When the assertion didn't start.
     NotYetStarted,
+    /// When the assertion is correct but cannot be processed
+    /// due to an unexpected reason.
+    Unprocessable(UnprocessableReason),
+}
+
+pub struct Message(String);
```

Comment:
See if we can create a `Log` trait instead so each DSL implement it with a `log` function.

The introduction of `Part::JsonPath` with `Hand::Compound` changes the way we log assertion results.

Reply:
See #38